src/history/historyTracker.ts

```diff
@@ -100,7 +100,7 @@ class HistoryStep {
     cursorEnd?: Position[] | undefined;
     marks?: IMark[];
   }) {
-    this.changes = init.changes = [];
+    this.changes = init.changes || [];
```

Comment:
Was this just a bug? Could this be the cause for the undo issues we've been experiencing?

Reply:
Yes, this was a bug that was preventing me from creating the new `HistoryStep` near the end of the function. Any list of changes passed in would always be overwritten as an empty array. I'm not aware of the existing undo issues, but it's possible.